operators/pkg/exmagnt/options.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exmagnt
+
+import (
+	"errors"
+	"flag"
+	"net/http"
+	"strings"
+
+	"github.com/ryanuber/go-glob"
+	"k8s.io/klog/v2"
+
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/utils/restcfg"
+)
+
+type options struct {
+	AllowedIPs       string
+	TemplatesNS      string
+	BasePath         string
+	ListenerAddr     string
+	PrintRequestBody bool
+}
+
+// Options object holds all the instanceset parameters.
+var Options options
+
+// Initialize flags and associate each parameter to the given options object.
+func (o *options) Init() {
+	flag.StringVar(&o.ListenerAddr, "address", ":8888", "[address]:port of the landing server")
+	flag.StringVar(&o.TemplatesNS, "templates-namespace", "", "Namespace of CrownLabs Templates that will be listed")
+	flag.StringVar(&o.AllowedIPs, "allowed-ips", "", "Comma separated list of IPs that are allowed to create new instances")
```

Comment:
Could probably leverage something like this to automatically perform IP validation: https://github.com/liqotech/liqo/blob/master/pkg/utils/args/cidrlist.go

Reply:
a eh actually made it myself before seeing this link...

Reply:
The outcome is slightly different, as with the other approach it is the library itself that returns the error during the flag parsing operation (printing also the appropriate message), instead of doing that in a post processing step. Still it is also fine as you did for this use-case